src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestIdentitiesDataStore.java

```diff
@@ -0,0 +1,129 @@
+package com.minecolonies.coremod.colony.requestsystem.data;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.colony.requestsystem.data.IRequestIdentitiesDataStore;
+import com.minecolonies.api.colony.requestsystem.factory.FactoryVoidInput;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.TypeConstants;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_LIST;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_REQUEST;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_TOKEN;
+
+/**
+ * A default implementation of the {@link IRequestIdentitiesDataStore} interface.
+ */
+public class StandardRequestIdentitiesDataStore implements IRequestIdentitiesDataStore
+{
+    private IToken<?>                     id;
+    private final BiMap<IToken<?>, IRequest<?>> map;
+
+    public StandardRequestIdentitiesDataStore(
+      final IToken<?> id,
+      final BiMap<IToken<?>, IRequest<?>> map) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)